ford/sourceform.py

```diff
@@ -1078,6 +1083,19 @@ def correlate(self, project: Project) -> None:
                         if hasattr(intr.procedure, "retvar"):
                             proc.retvar = intr.procedure.retvar
                         proc.proctype = intr.procedure.proctype
+                # Some module procs are from procedures implemented withen a generic interface
```

Comment:
Ok, it took me a few minutes, but I think I understand what situation this covers. Is it this:

```fortran
interface foo
  ! module procedure interface
  module function foo_impl(arg) result(res)
    integer :: arg, res
  end function
end interface
...
! module procedure implementation
module procedure foo_impl
end procedure
```

and here `proc` is the module procedure implementation and `intr` is its interface. The existing code only covers the case where the `interface` is not generic, is that right?

Given how similar these two blocks are, could we pull out a helper function to reduce the duplication?

Reply:
Yes, this is what it covers, and it's tested via test_project->test_generic_interface_inherited_attrs.